OneSignalSDK/onesignal/src/main/java/com/onesignal/OutcomeEventsController.java

```diff
@@ -175,15 +231,64 @@ public void run() {
         }
     }
 
-    void sendOutcomeEvent(@NonNull String name, @NonNull Bundle params) {
-        //TODO when backend changes are done
+    /**
+     * Get the unique notifications that have not been sent before with the current unique outcome name
+     */
+    private JSONArray getUniqueNotificationIds(String name, JSONArray notificationIds) {
+        boolean hasNoUniqueNotificationIds = hasNoUniqueOutcomeNotificationIds(name, notificationIds);
+        if (hasNoUniqueNotificationIds)
+            return null;
+
+        return getUniqueOutcomeNotificationIds(name, notificationIds);
     }
 
-    void sendOutcomeEvent(@NonNull String name, @NonNull String value) throws OutcomeException {
-        if (value.isEmpty())
-            throw new OutcomeException("Value must not be empty");
+    /**
+     * Validate whether or not the JSONArray of notificationIds has ids that have not been sent with the specific unique outcome name
+     */
+    private boolean hasNoUniqueOutcomeNotificationIds(String uniqueName, JSONArray notificationIds) {
```

Comment:
is this being used on other place? maybe it can return the JSONArray and avoid two fors